### PR TITLE
Select accessibility updates

### PR DESCRIPTION
**Pull request by jtanner**
* Use `Listbox` for the options' accessibility
* Handle `esc` to close
* Handle `enter`/`space` to open (when Select is focused)
* Give visual indication that the component is focused
* Use css hover instead of state
* Fix propagation bug with `Listbox`

Here's a demo of using only the keyboard to interact with `Select`:

ezgif com-video-to-gif

### CODE REVIEW

File: src/components/Select.js [hunk 1]
```diff
@@ -35,95 +37,43 @@ class Select extends React.Component {
...
-      };
-    } else {
-      return null;
+  _handleKeyDown = (e) => {
+    switch (keycode(e)) {
+      case 'esc':
+        e.preventDefault();
+        e.stopPropagation();
+        this._close();
+        this.component.focus();
```
Comment (iheartrachie) [inline src/components/Select.js]: Is this needed since _close() calls focus?

Comment (jtanner) [inline src/components/Select.js]: No, good catch, I meant to remove that 👍

Comment (jtanner) [inline src/components/Select.js]: I removed it.

**Comment by iheartrachie** [comment]
:shipit:

**Comment by redperk** [comment]
:shipit:

**Pull request merged**